File: src/WorkflowShell/git/GitAddDot.rb
require_relative '../models/Command'

class GitAddDot < Command
  @command_string
  @command_description
  @command_usage

  def initialize
    @command_string = 'gad'
    @command_description = 'Stages all files and folders in the current directory.'
    @command_usage = @command_string
  end

  def run_command(arguments)
    # Ask the Command class to parse the basic options like help and verbosity
    parser_components = parse_options(arguments, true)

    command = "git add ."
    run_shell_command(command, parser_components.basic_options.verbose)
  end
end

File: src/WorkflowShell/git/GitAddDot.ts
import Command from "../models/Command";

export default class GitAddDot extends Command {

  static getString(): string {
    return "gad";
  }

  static getDescription(): string {
    return "Stages all files and folders in the current directory.";
  }

  getUsage(): string {
    return GitAddDot.getString();
  }

  async run(args: any) {
    const ok: boolean = await super.run(args);
    if (!ok) {
      return false;
    }

    const baseCommand: string = "git add .";
    super.execute(baseCommand);
    return true;
  }
}
